### DISCUSSION TITLE
GSoD2021

### DISCUSSION THREAD

**Discussion by javipus** [pytorch/ignite#1959] [Q&A]
# Summary

This is my expression of interest for Google's Season of Docs program.

The project statement is still a draft and will be completed later. This is in part because some details about the proposal are unclear to me. I thought it would be beneficial for everyone to post them in public in case other writers have similar questions.

# Personal information

## Basic info

- My email: [javier.prieto.set@gmail.com](mailto:javier.prieto.set@gmail.com)
- My [CV](https://drive.google.com/file/d/1Ia_XJRJcs-n7Dp4f1zkYB1rmUPyMwamm/view?usp=sharing)
- My [GitHub profile](https://github.com/javipus/)

### UMA
I am currently doing technical support for the [Universal Market Access](https://umaproject.org/) (UMA) project, a startup developing tools to streamline access to financial derivatives on the Ethereum blockchain. As part of my involvement in the project, I have developed some drafts for financial products built using their protocol. These require a combination of technical specifications and financial engineering.

- [VTI UMIP](https://github.com/UMAprotocol/UMIPs/blob/231444334e96cc1d829ce7a824db1dc7215ee692/UMIPs/umip-draft_vti_priceID.md) proposes an implementation of the Vanguard Total Stock Market Index using UMA's infrastructure
- [uFYI](https://discourse.umaproject.org/t/forever-young-index/921) designs a market cap-weighted index tracking publicly traded biotechnology companies focused on longevity and anti-aging research
- [uIG](https://discourse.umaproject.org/t/impermanent-gains/918) is a product that helps liquidity providers on Uniswap-like automated market makers hedge against price volatility

### Academic writing
- I have published two proceedings papers on accelerator physics ([1](https://accelconf.web.cern.ch/ipac2018/doi/JACoW-IPAC2018-TUPAF050.html), [2](https://accelconf.web.cern.ch/ipac2018/doi/JACoW-IPAC2018-TUPAF052.html))
- I wrote a BSc thesis on [computational neuroscience](https://github.com/javipus/spikingNets/blob/master/thesis.pdf), and (yet another) BSc thesis on [the role of Lie theory in particle physics](https://github.com/javipus/tfg-lie/blob/master/thesis/Javier_Prieto_Prieto_tfg_lie_final.pdf).

### Blog
I have a (seldom updated) [blog](https://2e10e120.tumblr.com/) where I compile thoughts on things I read, mostly focusing on academic research I find interesting.

### Metaculus
I have authored [several questions](https://www.metaculus.com/questions/?search=author:2e10e122) on Metaculus, a forecasting platform that aggregates user predictions about the likelihood of future events. The questions are written for a lay audience, provide necessary background about the topic at hand, and establish clear criteria to adjudicate their resolution.

### EA
I used to be active in my local [Effective Altruism](https://www.effectivealtruism.org/) chapter, where I gave some talks:
- On [cash transfers](https://docs.google.com/presentation/d/1LQx-h6E1X0tIdmder0ZGoamM8qSouzNPEGChMGn01cw/edit?usp=sharing) for poverty alleviation
- On [charity effectiveness evaluation](https://docs.google.com/presentation/d/1qZ6SjPqqiBcq_XRpZ_BZ9MYBzQ-R8RLra46UHev0zZk/edit?usp=sharing) as done [GiveWell](https://www.givewell.org/)
- On [wild animal suffering](https://docs.google.com/presentation/d/1dGSkoUqFv2bQkdcvLOXbIDXlquJBIUgW81qYDNDeN-c/edit?usp=sharing)

# Project statement (in progress)

## Project plan

I usually need to get my hands dirty to learn anything. Therefore, I expect to audit the existing documentation by following every step to the letter and writing down any hiccups I find along the way. This will serve as the backbone for any documentation changes. Analogously, I expect to start developing new tutorials by attempting to do the required tasks (cross-validation, hyperparameter tuning) and documenting every step with care.

These experiences will inform the friction log. However, since I am fully aware that my perspective is far from representative, I expect input from users will be key at this step. User feedback will be incorporated in several ways. Firstly, tasks will be prioritized using [GitHub issues labeled as questions](https://github.com/pytorch/ignite/issues?q=is%3Aopen+is%3Aissue+label%3Aquestion), with preference given to open issues, as the closed ones may not be relevant anymore. Secondly, a user survey could be conducted to identify pain points, ideally with some kind of incentive in place (monetary or otherwise) to make participation more attractive.

## Measuring success

I strongly suggest A/B testing all the documentation developed during the project so that we don't fool ourselves and confuse correlation for causality. The details of the study design would depend exactly on the kind of docs we end up developing and whether they contain more new stuff or enhancements to previously existing material. At any rate, the release of new documentation should be staged in such a way that parts of it can be kept as controls for some of time in order to measure its true causal effect.

**Comment by vfdev-5** [comment]
@javipus thanks for your interest in our project and providing your CV, portfolio and a WIP on the project statement !

To answer your questions:

> Question: Will volunteers/users be compensated to test these tasks? Or is the friction log supposed to be developed based on my observations alone? I think this point isn't super clear in the proposal and a more detailed explanation would help a lot!

The friction log can be developed based on: 1) existing github issues labeled with "question", 2) users and volunteers (we can pay them if it can help to get more feedback), 3) our subjective feedback by dev team and 4) your observations especially when you'll dive into existing docs, tutorials etc.

> Question: Have you considered A/B testing success metrics? Say we develop N tutorials. It'd make sense to release N/2 first, keep rest as control (for a while). Otherwise we run the risk of confusing correlation with causality. When this data is in we can decide whether to make any changes to the remaining half.

Yes, this can be a metric too. We are open to any reasonable propositions on that as well.

**Reply by javipus** [reply]
Thanks for the thorough reply! I have edited the project statement to reflect this new info.
